tinygrad/tensor.py

```diff
@@ -388,7 +388,8 @@ def conv2d(self, weight:Tensor, bias:Optional[Tensor]=None, groups=1, stride=1,
   def dot(self, w:Tensor) -> Tensor:
     x = self.reshape(*self.shape[0:-1], 1, self.shape[-1])
     w = w.reshape(*w.shape[0:-2], 1, w.shape[-2], w.shape[-1]).transpose(-1, -2)
-    return (x*w).sum(-1).reshape(*x.shape[0:-2], -1)
+    r = (x*w).sum(-1)
+    return r.reshape((*r.shape[:-2], r.shape[-1]) if len(self.shape) == 1 else r.shape)
```

Comment:
`r if len(self.shape) == 1 else r.reshape((*r.shape[:-2], r.shape[-1]))` is the same? Avoid the r.reshape(r.shape)